code_challenges/Account_Generator.py

```diff
@@ -0,0 +1,60 @@
+import random
+
+
+def account_generator():
+    student_names = []
+    student_id_numbers = []
+    student_email_addresses = []
+    email_name = []
+
+    for i in range(int(input("enter how many students you have: "))):
+        student_names.append(input("Enter student name: "))
+
+    for name in student_names:
+        student_id_number = random.randint(111111, 999999)
+        student_id_numbers.append(student_id_number)
+        while student_id_number in student_id_numbers:
+            student_id_number = random.randint(111111, 999999)
+        student_name = name.split(" ")
+        if len(student_name) > 2:
+            email_name.append(student_name[0][0] + student_name[1][0] + student_name[2])
+        else:
+            email_name.append(student_name[0][0] + student_name[1])
```

Comment:
If you're handling the 3 long case it might make sense to handle the 1 long case too.

Reply:
what do you mean by that?

Reply:
`if len(student_name) > 2:` is true for `"William Ely Paysinger".split(" ")`
else would happen for both `"William Paysinger".split(" ")` or `"Ely".split(" ")`